### DISCUSSION TITLE
keep track of deltaTicks will endup overflowing?

### DISCUSSION THREAD

**Discussion by rudiHammad** [axmolengine/axmol#1245] [General]
Hi,
I was wondering if it would be okey to keep track of the game elapsed time using the deltaTicks. For instance, say we have this pseudo code:
`
void Foo::init()
{
    float timePassed = 0.0;
}

void Foo::update(float delta)
{
    this timePassed += delta;
    //.....more
}
`
A float variable does overflow with 3.40282e+038.  I haven't done the exact calculation, but that means that it would take almost a year of this loop to be running nonstop to overflow. 
Being new to game development I was wondering if it is a bad practice to do that or if it is okey, considering that it is very unlikely to have a game loop running a year (plus we could reset timePassed if a new level loads, the player dies, etc...) .
I know it might be a very basic question, but I am new to game dev and I would like to know what experienced game devs think.

Thanks,
R

**Comment by DelinWorks** [comment]
Unless you want to build a real-time online server with 100% uptime and an elapsed delta time that doesn't reset during an entire year then yeah go ahead and just use `timePassed += delta`. Otherwise, keep in mind that floats lose accuracy over time, while you're not gonna overflow until a year, you might reach that one year mark faster or slower by 10 or more days, storing time with high accuracy isn't recommended with floats, if you REALLY need that accuracy then you can instead use [std::chrono::steady_clock](https://en.cppreference.com/w/cpp/chrono/steady_clock), you can store the time at the start of the game and keep subtracting it from the current time, and with that cast the results to a `std::chrono::duration<float>`, this way you're not using floats directly to store your time but instead you pre-calculate the results accurately and then get a float out of it.

**Comment by rudiHammad** [comment]
Cool, thanks for the explanation. Make sense.
